platform/admin/core/admin-core-appservice/src/main/java/org/codice/ddf/admin/application/service/migratable/BundleManager.java

```diff
@@ -0,0 +1,95 @@
+/**
+ * Copyright (c) Codice Foundation
+ *
+ * <p>This is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.admin.application.service.migratable;
+
+import com.google.common.annotations.VisibleForTesting;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import org.osgi.framework.BundleContext;
+import org.osgi.framework.FrameworkUtil;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/** Utility class capable of managing bundles during export or import. */
+public class BundleManager {
+  private static final Logger LOGGER = LoggerFactory.getLogger(BundleManager.class);
+
+  private final BundleProcessor processor;
+
+  @VisibleForTesting
+  BundleManager(BundleProcessor processor) {
+    this.processor = processor;
+  }
+
+  public BundleManager() {
+    this(new BundleProcessor());
+  }
+
+  /**
+   * Exports all bundles in memory into their corresponding Json format.
+   *
+   * @return a list of the Json format for all bundles in memory
+   */
+  public List<Map<String, Object>> exportBundles() {
+    return Stream.of(processor.listBundles(getBundleContext()))
+        .map(JsonUtils::toJson)
+        .collect(Collectors.toList()); // preserve order
+  }
+
+  /**
+   * Import the specified bundles.
+   *
+   * <p>The implementation will loop until it fails to find a bundle in memory or it fails to change
+   * the state of a bundle or again until the state of the bundles in memory matches the state of
+   * the bundles on the original system.
+   *
+   * @param report the report where to record errors
+   * @param jbundles the bundles to import
+   * @return <code>true</code> if all were imported successfully; <code>false</code> otherwise
+   */
+  public boolean importBundles(ProfileMigrationReport report, List<JsonBundle> jbundles) {
+    final BundleContext context = getBundleContext();
+    final TaskList tasks = newTaskList(report);
+
+    // loop until we can determine that all bundles that should be started. stopped, installed, or
+    // uninstalled are or until we get an error
+    while (true) {
+      // populate the task list
+      if (!processor.processBundles(context, report, jbundles, tasks)) {
```

Comment:
❓ If you're having to put this comment, why not just rename the method to better describe what it's doing?